syntax = "proto3";

package ngms.accounts;

option go_package = "github.com/sloppycoder/ngms/accounts/api;api_pb";


import "google/protobuf/timestamp.proto";
import "google/api/annotations.proto";
import "google/protobuf/empty.proto";

service AccountService {
  rpc GetAccount (GetAccountRequest) returns (Account) {
    option (google.api.http) = {
      get: "/accounts/{account_id}"
    };
  }
}

message Account {
  string id = 1;
  string account_id = 2;
  string nickname = 3;
  string prod_code = 4;
  string prod_name = 5;
  string currency = 6;
  string servicer = 7;
  Status status = 8;
  google.protobuf.Timestamp status_last_updated = 9;
  repeated Balance balances = 10;

  enum Status {
    ACTIVE = 0;
    BLOCKED = 1;
    DORMANT = 2;
  }
}

message Balance {
  double amount = 1;
  Type type = 2; // balance type
  bool credit_flag = 3; // credit or debit balance
  google.protobuf.Timestamp last_updated = 4;

  enum Type {
    CURRENT = 0;
    AVAILABLE = 1;
  }
}

message GetAccountRequest {
  string account_id = 1;
}
